Not enough space in game folder to install modules on Mac

Background
----------
* Operating System: macos catalina
* CKAN Version: 1.31.2
* KSP Version: 1.12.5

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?** no


Problem
-------
**Describe the bug**
crash when download mod

<img width="485" alt="Screen Shot 2023-01-29 at 5 55 09 PM" src="https://user-images.githubusercontent.com/40470811/215366870-dc604725-1580-46b4-bb08-db3b4e191093.png">

**Steps to reproduce**
* Go to...
* Click on...
* See error

**Expected behavior**


**Screenshots (if applicable)**



**CKAN error code (if applicable):**

```
Unhandled Exception:                                                          │
CKAN.NotEnoughSpaceKraken: Not enough space in game folder to install modules!┘
Need to store 2.3 MiB to /Applications/ksp_osx, but only 0 B is available!     
Free up space on that device or change your settings to use another location.

  at CKAN.CKANPathUtils.CheckFreeSpace (System.IO.DirectoryInfo where, System.Int64 bytesToStore, System.String errorDescription) [0x00042] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ModuleInstaller.InstallList (System.Collections.Generic.ICollection`1[T] modules, CKAN.RelationshipResolverOptions options, CKAN.RegistryManager registry_manager, System.Collections.Generic.HashSet`1[System.String]& possibleConfigOnlyDirs, CKAN.IDownloader downloader, System.Boolean ConfirmPrompt) [0x00086] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.InstallScreen.Run (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme, System.Action`1[T] process) [0x00125] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.Toolkit.ConsoleScreen.LaunchSubScreen (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme, CKAN.ConsoleUI.Toolkit.ConsoleScreen cs, System.Action`1[T] newProc) [0x00000] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ModListScreen.ApplyChanges (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme) [0x00019] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ModListScreen.<.ctor>b__0_26 (System.Object sender, CKAN.ConsoleUI.Toolkit.ConsoleTheme theme) [0x00000] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.Toolkit.ScreenContainer.Interact (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme) [0x00038] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.Toolkit.ScreenContainer.Run (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme, System.Action`1[T] process) [0x00021] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ConsoleCKAN..ctor (CKAN.GameInstanceManager mgr, System.String themeName, System.Boolean debug) [0x00089] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ConsoleUI.Main_ (System.String[] args, CKAN.GameInstanceManager manager, System.String themeName, System.Boolean debug) [0x00005] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.ConsoleUi (CKAN.GameInstanceManager manager, CKAN.CmdLine.ConsoleUIOptions opts, System.String[] args) [0x00034] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.GameInstanceManager manager) [0x002df] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x00296] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x000e9] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: CKAN.NotEnoughSpaceKraken: Not enough space in game folder to install modules!
Need to store 2.3 MiB to /Applications/ksp_osx, but only 0 B is available!
Free up space on that device or change your settings to use another location.

  at CKAN.CKANPathUtils.CheckFreeSpace (System.IO.DirectoryInfo where, System.Int64 bytesToStore, System.String errorDescription) [0x00042] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ModuleInstaller.InstallList (System.Collections.Generic.ICollection`1[T] modules, CKAN.RelationshipResolverOptions options, CKAN.RegistryManager registry_manager, System.Collections.Generic.HashSet`1[System.String]& possibleConfigOnlyDirs, CKAN.IDownloader downloader, System.Boolean ConfirmPrompt) [0x00086] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.InstallScreen.Run (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme, System.Action`1[T] process) [0x00125] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.Toolkit.ConsoleScreen.LaunchSubScreen (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme, CKAN.ConsoleUI.Toolkit.ConsoleScreen cs, System.Action`1[T] newProc) [0x00000] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ModListScreen.ApplyChanges (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme) [0x00019] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ModListScreen.<.ctor>b__0_26 (System.Object sender, CKAN.ConsoleUI.Toolkit.ConsoleTheme theme) [0x00000] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.Toolkit.ScreenContainer.Interact (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme) [0x00038] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.Toolkit.ScreenContainer.Run (CKAN.ConsoleUI.Toolkit.ConsoleTheme theme, System.Action`1[T] process) [0x00021] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ConsoleCKAN..ctor (CKAN.GameInstanceManager mgr, System.String themeName, System.Boolean debug) [0x00089] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.ConsoleUI.ConsoleUI.Main_ (System.String[] args, CKAN.GameInstanceManager manager, System.String themeName, System.Boolean debug) [0x00005] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.ConsoleUi (CKAN.GameInstanceManager manager, CKAN.CmdLine.ConsoleUIOptions opts, System.String[] args) [0x00034] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.RunSimpleAction (CKAN.CmdLine.Options cmdline, CKAN.CmdLine.CommonOptions options, System.String[] args, CKAN.IUser user, CKAN.GameInstanceManager manager) [0x002df] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.GameInstanceManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x00296] in <79ae9046bd9c43dabfd2f7be4b883034>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x000e9] in <79ae9046bd9c43dabfd2f7be4b883034>:0
```

--- a/Core/Extensions/IOExtensions.cs
+++ b/Core/Extensions/IOExtensions.cs
@@ -48,13 +48,18 @@ namespace CKAN.Extensions
         /// that's a prefix of the dir's FullName.
         /// </summary>
         /// <param name="dir">Any DirectoryInfo object</param>
-        /// <returns>The DriveInfo associated with this directory</returns>
+        /// <returns>The DriveInfo associated with this directory, if any, else null</returns>
         public static DriveInfo GetDrive(this DirectoryInfo dir)
-            => DriveInfo.GetDrives()
-                        .Where(dr => dr.RootDirectory == dir
-                                     || dr.RootDirectory.IsAncestorOf(dir))
-                        .OrderByDescending(dr => dr.RootDirectory.FullName.Length)
-                        .FirstOrDefault();
+            => Platform.IsMono
+                // Mono's DriveInfo.GetDrives doesn't return mounted filesystems, so we
+                // can't get the drive for a dir on Linux or Mac
+                ? null
+                : DriveInfo.GetDrives()
+                           .Where(dr => dr.IsReady
+                                        && dr.DriveType != DriveType.NoRootDirectory
+                                        && dr.RootDirectory.IsAncestorOf(dir))
+                           .OrderByDescending(dr => dr.RootDirectory.FullName.Length)
+                           .FirstOrDefault();
 
         /// <summary>
         /// A version of Stream.CopyTo with progress updates.

--- a/Core/GameInstanceManager.cs
+++ b/Core/GameInstanceManager.cs
@@ -572,7 +572,7 @@ namespace CKAN
                 else
                 {
                     // Make sure we can access it
-                    var bytesFree = new DirectoryInfo(path).GetDrive().AvailableFreeSpace;
+                    var bytesFree = new DirectoryInfo(path).GetDrive()?.AvailableFreeSpace;
                     Cache = new NetModuleCache(this, path);
                     Configuration.DownloadCacheDir = path;
                 }

--- a/GUI/CKAN-GUI.csproj
+++ b/GUI/CKAN-GUI.csproj
@@ -1277,7 +1277,7 @@
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
     <EmbeddedResource Include="Properties\Resources.ko-KR.resx">
-      <LogicalName>CKAN.Properties.Resources.ko-KR.resources</LogicalName>
+      <LogicalName>CKAN.GUI.Properties.Resources.ko-KR.resources</LogicalName>
       <SubType>Designer</SubType>
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>

